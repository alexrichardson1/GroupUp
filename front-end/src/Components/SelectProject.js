import React, { useState } from "react";
import { Link } from "react-router-dom";
import DropdownMenu from "./DropdownMenu";
import years from "./years.json";
import projects from "./projects.json";

const SelectProject = (props) => {
  const [value, setValue] = useState(null);
  const [proejctValue, setProjectValue] = useState(null);

  return (
    <div>
      <body>
        Please give us more details about your project so we can help you find
        the best team for you.
      </body>
      <h1>Imperial</h1>
      Year{" "}
      <DropdownMenu
        promt="Select year"
        options={years}
        value={value}
        id="id"
        label="id"
        onChange={(val) => setValue(val)}
      />
      Project{" "}
      <DropdownMenu
        promt="Select project"
        options={projects}
        value={proejctValue}
        id="name"
        label="name"
        onChange={(val) => setProjectValue(val)}
      />
      <Link to="/listings">
        <button className="btn">Find your group</button>
      </Link>
    </div>
  );
};

export default SelectProject;
