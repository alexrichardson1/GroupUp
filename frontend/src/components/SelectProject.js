import React from "react";
import { Dropdown } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import NavBar from "components/NavBar";
import axios from "axios";
import { useState, useEffect } from "react";

const SelectProject = () => {
  const history = useHistory();
  const handleProjectClick = (id) => {
    history.push("/listing/" + id);
  };

  const [projects, setProjects] = useState([]);

  useEffect(() => {
    const getProjects = async () => {
      await axios
        .get("http://localhost:5000/project")
        .then((res) => {
          const projects = res.data;
          console.log(projects);
          setProjects(projects);
        })
        .catch((error) => {
          console.log(error);
        });
    };
    getProjects();
  }, []);

  return (
    <div>
      <NavBar renderBool={[true, true, false, false]} create={false} />
      <h1>List of Competitions Available</h1>
      <Dropdown>
        <Dropdown.Toggle variant="success" id="dropdown-basic">
          Select Project
        </Dropdown.Toggle>

        <Dropdown.Menu>
          {projects.map((proj) => (
            <Dropdown.Item
              key={proj.id}
              onSelect={() => handleProjectClick(proj.id)}
            >
              {proj.name}
            </Dropdown.Item>
          ))}
          {
            //Add href="#/your-action" when action available
          }
        </Dropdown.Menu>
      </Dropdown>
      {/* <LinkContainer to="/listings">
        <Button>Show groups</Button>
      </LinkContainer> */}
    </div>
  );
};

export default SelectProject;
