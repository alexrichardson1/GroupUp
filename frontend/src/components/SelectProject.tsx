import { useState, useEffect } from "react";
import { Table, Button } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import NavBar from "components/NavBar";
import axios from "axios";
import { config } from "Constants";
import { ProjectT } from "types/types";

const SelectProject = () => {
  const [projects, setProjects] = useState<ProjectT[]>([]);

  useEffect(() => {
    const getProjects = async () => {
      await axios
        .get(`${config.API_URL}/project`)
        .then((res) => {
          const projects = res.data;
          console.log(projects);
          setProjects(projects);
        })
        .catch((error) => {
          console.log(error);
        });
    };
    document.title = "Select a Hackathon";
    getProjects();
  }, []);

  return (
    <div>
      <NavBar renderBool={[true, true, false, false]} create={false} />
      <h1>List of Competitions Available</h1>

      <Table striped bordered hover size="sm">
        <thead>
          <tr>
            <th>Hackathon Name</th>
            <th>Description</th>
            <th>Start Date</th>
            <th>Length (hours)</th>
            <th>Link</th>
          </tr>
        </thead>
        <tbody>
          {projects.map((proj) => (
            <tr>
              <td>{proj.name}</td>
              <td>{proj.description}</td>
              <td>{proj.date.substring(0, 10)}</td>
              <td>{proj.hours}</td>
              <td>
                <LinkContainer to={`/listing/${proj.id}`}>
                  <Button>View Listings</Button>
                </LinkContainer>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
      <LinkContainer to="/createHackathon">
        <Button className="advertBtn">Add my Hackathon</Button>
      </LinkContainer>
    </div>
  );
};

export default SelectProject;
