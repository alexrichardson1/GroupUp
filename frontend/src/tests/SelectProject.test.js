import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import SelectProject from "components/SelectProject";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SelectProject />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
