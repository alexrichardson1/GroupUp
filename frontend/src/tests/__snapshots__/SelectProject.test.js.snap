// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navBar navbar navbar-expand-lg navbar-light fixed-top"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "color": "white",
        }
      }
    >
      <img
        alt="group up logo"
        className="logoImage"
        href="/home"
        onClick={[Function]}
        src="https://i.imgur.com/w92t1xH.png"
      />
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="mr=auto navbar-nav"
        onKeyDown={[Function]}
      >
        <span
          className="navbar-brand"
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
           
        </span>
        <a
          className="nav-link"
          data-rb-event-key="/home"
          href="/home"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Home
          <img
            alt="drop down arrow"
            className="processImage"
            src="/right-caret.svg"
          />
        </a>
        <span
          className="navbar-brand"
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Select Hackathon
        </span>
      </div>
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="ml-auto navHelp navbar-brand"
          href="/help"
          id="collasible-nav-dropdown"
          onClick={[Function]}
          title="Help"
        >
          <img
            alt="help"
            className="splitterImage"
            src="/info-circle.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <h1>
    List of Competitions Available
  </h1>
  <table
    className="table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Hackathon Name
        </th>
        <th>
          Description
        </th>
        <th>
          Start Date
        </th>
        <th>
          Length (hours)
        </th>
        <th>
          Link
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <a
    className="advertBtn btn btn-primary"
    href="/createHackathon"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Add my Hackathon
  </a>
</div>
`;
